### DISCUSSION TITLE
关于如何更好地传播的建议

### DISCUSSION THREAD

**Discussion by shuiRong** [LGBT-CN/LGBTQIA-In-China#83] [Show and tell]
考虑制作一个看上去很漂亮的Github Badge，然后呼吁大家把它加入到自己的项目中。毕竟一个漂亮好看的Badge，谁不愿意加呢？

当然，点击该Badge会跳转到某链接（比如你们的项目主页）。

这是[996.icu](https://github.com/996icu/996.ICU)的Badge，但还不够好看。
![image](https://user-images.githubusercontent.com/16076993/79591313-e6335000-810a-11ea-9172-a8d844df13db.png)

更多[Badge](https://github.com/Naereen/badges)参考。

此外，如果有能力，也可以考虑写一个工具（Github App或者其他什么载体可以实现我不确定）来自动给项目提交PR来加上该Badge。或许先搜搜社区有没有这样的工具？

工具预想是这样的：A觉得这个Badge不错，然后输入自己的项目名称，点击确定按钮。程序就会自动Fork该项目并且在README中加上Badge，然后自动提交PR。A只需要点击下合并，就会看到该Badge出现在自己的项目中。

**Comment by kecrily** [comment]
好的，我尝试去搞一下。

**Comment by kecrily** [comment]
> 好的，我尝试去搞一下。

@shuiRong 其实感觉是没多少愿意加入这种用处不大的徽章的，另外 LGBTQIA 群体中愿意公开身份人数应该也不多。
徽章的设计可能要晚点再进行，目前自己不会设计也找不到设计。
最后我刚刚开了一个 **[电报群](https://t.me/LGBTCN)**，方便沟通，可以考虑加入一下。

**Comment by Mikedu1988** [comment]
同情 支持
但是自身并不是群体的一员，也不能做等多。
repo突然出现在我的github首页有点意外。
加油。

**Comment by ghost** [comment]
@KevinZonda 首先，网站我认为应该多加一些语言，方便留学生的浏览，以及港澳台同胞的浏览（我已经添加了繁体中文和英文的页面，可以看一下[拉取请求](https://github.com/LGBT-CN/LGBTQIA-in-China/pull/33)）。然后，我觉得你应该向[996.ICU](https://github.com/996icu/996.ICU)和[wuhan2020](https://github.com/wuhan2020/wuhan2020)学习一下，看看他们是如何构建、运作的。
@scvoet 我建议你换成slack，因为telegram需要翻墙，翻墙软件不是谁都有的。

**Comment by KevinZonda** [comment]
@zrzjohn Your PR has been reviewed. Plz resolve problems then I will merge it. :)

**Comment by KevinZonda** [comment]
@zrzjohn Hi there, remaining suggestions I will communicate with @scvoet ! Thanks a lot!

**Comment by BigCoke233** [comment]
I tried to make a badge for this project, but it doesn't seems to work. I want to make it more like a symbol for an organization about LGBTQIA rights which is marked by six-colored rainbow. Unfortunately, a tiny badge with low pixels cannot hold six colors well enough and the text is damaged. It looks like...

![](https://cdn.jsdelivr.net/gh/BigCoke233/image-hosting@2020.10.7/issues%20%26%20pr/LGBTQIA-in-China-issue-12/rainbow-badge-1.png)
![](https://cdn.jsdelivr.net/gh/BigCoke233/image-hosting@2020.10.7/issues%20%26%20pr/LGBTQIA-in-China-issue-12/rainbow-badge-2.png)

SVG won't be a better choice, it's hard to draw a rainbow by SVG. So I am considering other solutions like this:

![](https://img.shields.io/badge/style-flat--square-green?logo=appveyor&style=flat-square)

We can put the logo ahead the text, but the current logo is a little complex, which makes it hard to recognize in a tiny image. Maybe we need a new, simple logo or something.

Or we can just use a simple badge without any decorations:

![](https://img.shields.io/badge/link-LGBTQIA--in--China-blueviolet?style=flat-square)

**Comment by KevinZonda** [comment]
@BigCoke233 I think we need a small logo which can suit for such things

**Comment by BigCoke233** [comment]
By the way, I see @shuiRong suggest that we can create a tool (or application?) to help project manager add badge into his/her repository. But it's a little hard because we have no idea about where the manager want to put the badge, top of the doucument? below the title? He/She may already have some other badges, shouldn't the new badge be placed in the same line? How does the app figures out where the old badges are placed? If the app just put the badge in a fixed position (like the top of document), the manager has to edit README file and move the badge where he/she wants it to be. I don't think it would help anyway.

---

顺带一提，我看见 @shuiRong 提出我们可以做一个工具来帮助仓库管理者为他的仓库添加徽章。但这有一点不好办，因为我们并不知道管理者想要把徽章放在什么地方，文档顶部？标题下面？他/她可能已经有了一些其他的徽章，新的徽章不也该和它们放在一起吗？这个工具要怎么才能知道这些旧的徽章在哪里？如果这个工具就只是把徽章随便放在一个固定的位置，管理者还需要重新编辑 README 文件来移动这个徽章到 TA 想要的地方。我不觉得这能帮上忙。

**Comment by Michael18811380328** [comment]
貌似近期 github 被墙了，有木有更好的传播方式呢？

**Reply by ghost** [reply]
GitHub现在有时被墙，有时又能正常访问。真不知道方先生搞得什么算法。
而且基本上都是假墙，hosts只要更新快一点都能上去。

**Comment by KevinZonda** [comment]
> 貌似近期 github 被墙了，有木有更好的传播方式呢？

我们可以试试看用Cloudflare，但是最好的方法还是购置自己的机器。
/cc @scvoet

**Comment by kecrily** [comment]
@KevinZonda 啊，我没听懂

**Comment by KevinZonda** [comment]
Hi there. Thanks for coool jobs have been done. But I don't think it's a good decision that move such to here:

1. Discussion doesn't have notify number - which may downgrade the attraction of the topic
2. For common habits, people may use issues more often than discussion - same result will be caused as above.

**Reply by kecrily** [reply]
> 
> 
> /cc @scvoet

We can't turn discussions back to issues anymore, I personally think it's just a matter of habit that people don't read discussions, it's still too new. We just need to take a little time to wait.

**Comment by Michael18811380328** [comment]
翻译问题，可以在公共的翻译平台做一下，例如 transfix 上面可以实现多种国际语言，这样依靠大家的力量才能完善。现在网站是一个单页面，如果使用一些框架（例如 django 和 flask 后端框架），可以实现语言国际化切换，不需要每一种语言维护单独的界面。

**Reply by KevinZonda** [reply]
我们可以写一个类似于generator的东西

**Reply by KevinZonda** [reply]
我感觉这个项目其实并不是很想使用后端，当然，问问percy
/cc @scvoet

**Reply by KevinZonda** [reply]
我在实现 Kvm 时候有一个非常类似的工具 源码可以复用改改就行 我马上开个repo？
/cc @scvoet

**Reply by KevinZonda** [reply]
I'm developing a tool called Kvm, which is now in preview, hope it will help <https://github.com/LGBT-CN/Kvm>

**Reply by LLLgoyour** [reply]
crowdin已经足够强大，可以考虑

> 我们可以写一个类似于generator的东西

**Comment by KevinZonda** [comment]
LGBT-CN's website has been migrated into Kvm in order to help translation.

See more:
https://github.com/LGBT-CN/lgbt-cn.github.io
https://github.com/LGBT-CN/Kvm

**Reply by Michael18811380328** [reply]
well down

**Comment by lzhbhlrPython** [comment]
啊……我已经拉着我的小伙伴署名了，虽然我们不是这个圈子的成员，但我们很愿意支持这个人群平权，另外GitHub之外或许可以试试国内的coding page或者在港澳台地区有服务器且比较稳定的Vercel/Netlify。badge确实挺有风险的（

**Reply by KevinZonda** [reply]
我准备将LGBT-CN的网站迁移至KevinZonda Network！

**Comment by KevinZonda** [comment]
考虑过，但是实际上在一些产品的实践证明这种东西并不符合大多数实践场景。

19 Nov 2021, 05:56 by ***@***.***:
